iFixit/react-commerce
# Hero with background frontend changes

closes #1170 

Mockup: https://www.figma.com/file/jRG4EyQXRDMGfyLqut08KP/iFixit---Working-files?node-id=8713%3A141297

This PR introduces an `HeroWithBackground` component for product lists.
It substitutes the default hero when a valid `heroImage` is provided.
The `brandLogo` is optional. If provided also a `brandLogoWidth` must be provided

### Note
The margins around the component are different from the mockup. 
This is due to the fact that here we are only introducing the new component in the existing structure.
Issue https://github.com/iFixit/design/issues/510 will be divided in subtasks and will be gradually integrated in other PRs.

### QA

1. Visit [Vercel preview](https://react-commerce-git-implement-hero-with-background-ifixit.vercel.app/Parts)
2. Try adding a `heroImage` to a specific product list in [Strapi](https://implement-hero-with-background.govinor.com/admin)
3. Verify that the heroWithBackground correctly substitutes the default hero
